Backend/app/controllers/user.controller.js

```diff
@@ -317,27 +317,50 @@ exports.register = (req,res) => {
                     password,
                     password2
                 })
-                //objekt mit daten, die in DB geschrieben werden sollen wird gebaut
             } else {
-                const newUser = new User({
-                    username, email, password,token:uuid.v4()
-                });
+                User.findAll({
+                    where: {
+                        username: {
+                            [OP.like]: username
+                        }
+                    }
+                }).then(user => {
+                    //Überprüfung, ob der Username bereits existiert und Rückgabe einer Liste mit allen gefundenen usern
+                    if (user[0] && user[0].username) {
+                        errors.push({msg: 'Username is already used'})
+                    }
+                    if (errors.length > 0) {
+                        res.json(400, {
+                            errors,
+                            username,
+                            email,
+                            password,
+                            password2
```

Comment:
Hier bitte nicht alle Daten mit ans Frontend schicken, wenn es zu einem Fehler kommt, sondern nur die errors

Reply:
Done in the whole method